app/controllers/users/notification_settings_controller.rb

```diff
@@ -0,0 +1,46 @@
+module Users
+  class NotificationSettingsController < ApplicationController
+    before_action :raise_suspended
+    before_action :authenticate_user!
+    after_action :verify_authorized
```

Comment:
Should we verify authorization _before_ running the action instead of after? It feels like running it afterward would still perform the update but then return a `403 Forbidden` response. Am I misunderstanding?

Reply:
My suggestion came from the existing actions on `UsersController`:

https://github.com/forem/forem/blob/8963165e6ea52c47feda570950d46b85b1f75d8a/app/controllers/users_controller.rb#L1-L11

That's why I suggested copying the relevant ones to this new controller

Reply:
[nonblocking]

I still feel like this should be a `before_action` because authorizing after the transaction has been committed, at which point an unauthorized condition can't block the operation without a flux capacitor. 😄 

However, I've been completely unable to do anything nefarious with this locally because it requires an authenticated user session to be able to run the update anyway (since that's how it knows which user notification setting to update). Since authenticated users are always authorized to change their own notification settings and we need to be authenticated to get a notification setting anyway, do we actually need the authorization?

To be clear, I think it would make perfect sense if we still had the `id` in the URL and were selecting the record based on that param.